float32 posX
float32 posY
